test/apps/iodemo/io_demo.cc

```diff
@@ -948,6 +949,27 @@ class P2pDemoCommon : public UcxContext {
         validate(conn, msg, iomsg_size);
     }
 
+    BufferIov* prepare_recv_data_iov(size_t data_size)
+    {
+        BufferIov *iov = _data_buffers_pool.get();
+        iov->init(data_size, _data_chunks_pool, _iov_buf_filler,
+                  _iov_buf_filler, opts().validate);
+        return iov;
+    }
+
+    BufferIov* prepare_recv_data_iov(size_t data_size, const UcxAmDesc &data_desc)
+    {
+        BufferIov *iov = _data_buffers_pool.get();
+        if (!ucx_am_is_rndv(data_desc)) {
+            iov->init(data_size, ucx_am_get_data(data_desc));
+        } else {
+            iov->init(data_size, _data_chunks_pool, _iov_buf_filler,
+                      _iov_buf_filler, opts().validate);
```

Comment:
can we reuse prepare_recv_data_iov() here?
```
if (!ucx_am_is_rndv(data_desc)) {
    BufferIov *iov = _data_buffers_pool.get();
    iov->init(data_size, ucx_am_get_data(data_desc));
    return iov;
}

return prepare_recv_data_iov(data_size);
```

Reply:
done